File: framework/languages/java/samples/java/Bingo/Server/Play/src/main/java/systems/zlink/samples/bingo/server/play/adapters/zlink/handlers/BingoMatchQueue.java
package systems.zlink.samples.bingo.server.play.adapters.zlink.handlers;

public interface BingoMatchQueue {
    BingoMatchReservation reserve(
        String mode,
        String actorId,
        String preferredOwnerNodeRid,
        String newRoomId,
        int requiredPlayers);
}

File: framework/languages/java/samples/kotlin/Bingo/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/play/adapters/zlink/handlers/BingoMatchQueue.kt
package systems.zlink.samples.kotlin.bingo.server.play.adapters.zlink.handlers

interface BingoMatchQueue {
    fun reserve(
        mode: String,
        actorId: String,
        preferredOwnerNodeRid: String,
        newRoomId: String,
        requiredPlayers: Int,
    ): BingoMatchReservation
}
